ansible/cloud_providers/gcp_infrastructure_deployment.yml

```diff
@@ -0,0 +1,193 @@
+        - gen_gcp_template
+
+    - name: Generate GCP deployment manager template
+      template:
+        src: "../configs/{{ env_type }}/files/cloud_providers/{{cloud_provider}}_cloud_template.j2"
+        dest: "{{t_dest}}"
+      tags:
+        - gcp_infrastructure_deployment
+        - validate_gcp_template
+        - gen_gcp_template
+
+    # use command line 'gcloud' to validate template and deploy
+    - name: Login to Google Cloud SDK
+      command: >-
+        gcloud auth
+        activate-service-account "{{ gcp_account }}"
+        --key-file="{{ gcp_credentials_file }}"
+        --project="{{ gcp_project_id }}"
+      environment:
+        PATH: /usr/bin
+      tags:
+        - validate_gcp_template
+        - create_inventory
+        - must
+
+    - name: Get SSH public key
+      set_fact:
+        ssh_key: "~/.ssh/{{key_name}}.pem"
+        ssh_key_data: "{{lookup('file', '~/.ssh/{{key_name}}.pub')}}"
+      tags:
+        - validate_gcp_template
+        - must
+        - create_inventory
+
+    - name: Starting Google deployment-manager with template
+      environment:
+        PATH: /usr/bin
+      command: >-
+        gcloud deployment-manager deployments
+        create "{{ project_tag }}-base"
+        --config "{{ t_dest }}"
+      register: gcp_deploy
+      tags:
+        - gcp_infrastructure_deployment
+        - gcp_deployment_manager
+      until: gcp_deploy is succeeded
+      retries: 0
+
+    - debug:
+        var: gcp_deploy
+        verbosity: 2
+      tags:
+        - gcp_infrastructure_deployment
+
+    - name: Run infra-gcp-create-inventory Role
+      import_role:
+        name: infra-gcp-create-inventory
+
+    - name: Fetch DNS zone Info
+      gcp_dns_managed_zone_facts:
+        dns_name: '{{ cluster_dns_zone + "."}}'
+        project: "{{ gcp_project_id }}"
+        auth_kind: "{{ gcp_auth_type }}"
+        service_account_file: "{{ gcp_credentials_file }}"
+      register: gcp_managed_zone
+      tags:
+        - gcp_infrastructure_deployment
+      when:
+        - HostedZoneId != "none"
+
+    - name: Add delegation for NS to the main DNSZone
+      gcp_dns_resource_record_set:
+        project: "{{ gcp_project_id }}"
+        auth_kind: "{{ gcp_auth_type }}"
+        service_account_file: "{{ gcp_credentials_file }}"
+        managed_zone: "{{ gcp_managed_zone.resources[0] }}"
+        name: '{{ guid + "." + cluster_dns_zone + "."}}'
+        type: NS
+        ttl: 600
+        target:
+          - "{{ gcp_managed_zone.resources[0].nameServers[0] }}"
+          - "{{ gcp_managed_zone.resources[0].nameServers[1] }}"
+          - "{{ gcp_managed_zone.resources[0].nameServers[2] }}"
+          - "{{ gcp_managed_zone.resources[0].nameServers[3] }}"
+        state: present
+      tags:
+        - gcp_infrastructure_deployment
+      when:
+        - env_type != "ocp4-cluster"
+        - HostedZoneId != "none"
+
+    - name: Add bastion entry to the main DNSZone
+      gcp_dns_resource_record_set:
```

Comment:
same question

Reply:
same answer

Reply:
There is no clean way to include DNS records in the deployment manager template. I can include a third party script but then we would also need to branch that script and track changes to it. At least the way it is now it uses Ansible so its stateless.